pkg/pod/pod.go

```diff
@@ -283,6 +295,55 @@ func (builder *Builder) WaitUntilRunning(timeout time.Duration) error {
 	return builder.WaitUntilInStatus(corev1.PodRunning, timeout)
 }
 
+// WaitUntilHealthy waits for the duration of the defined timeout or until the pod is healthy.
+// An healthy pod is in running phase and optionally in ready condition.
+//
+// timeout is the duration to wait for the pod to be healthy
+// includeSucceeded when true, implies that pod in succeeded phase is running.
+// skipReadiness when true, to also checks that the podCondition is ready.
+// ignoreFailedPods when true, to Ignore failed pod with restart policy set to never.
+func (builder *Builder) WaitUntilHealthy(timeout time.Duration, includeSucceeded, skipReadinessCheck,
+	ignoreRestartPolicyNever bool) error {
+	statusesChecked := []corev1.PodPhase{corev1.PodRunning}
+
+	// Ignore failed pod with restart policy never. This could happen in image pruner or installer pods that
+	// will never restart. For those pods, instead of restarting the same pod, a new pod will be created
+	// to complete the task.
+	if ignoreRestartPolicyNever &&
+		builder.Object.Status.Phase == corev1.PodFailed &&
+		builder.Object.Spec.RestartPolicy == corev1.RestartPolicyNever {
+		glog.V(100).Infof("Ignore failed pod with restart policy never. Message: %s",
+			builder.Object.Status.Message)
+
+		return nil
+	}
+
+	if includeSucceeded {
+		statusesChecked = append(statusesChecked, corev1.PodSucceeded)
+	}
+
+	podPhase, err := builder.WaitUntilInStatuses(statusesChecked, timeout)
+
+	if err != nil {
+		glog.V(100).Infof("pod condition is not in %v. Message: %s", statusesChecked, builder.Object.Status.Message)
+
+		return err
+	}
+
+	if skipReadinessCheck || *podPhase != corev1.PodRunning {
```

Comment:
Shouldn't this just be `if skipReadinessCheck {` ? Does the podPhase being anything other than corev1.PodRunning indicate that the pod is ready? It seems like they check two different things.

Reply:
This is to support when a pod in succeeded phase. In that case, the pod is considered healthy as it has completed successfully.

Reply:
Fixed